modules/light-clients/08-wasm/types/store.go

```diff
@@ -123,3 +129,46 @@ func (s storeAdapter) Iterator(start, end []byte) wasmvmtypes.Iterator {
 func (s storeAdapter) ReverseIterator(start, end []byte) wasmvmtypes.Iterator {
 	return s.parent.ReverseIterator(start, end)
 }
+
+func getClientID(clientStore storetypes.KVStore) (string, error) {
+	upws, isUpdateProposalWrappedStore := clientStore.(updateProposalWrappedStore)
+	if isUpdateProposalWrappedStore {
+		// if the clientStore is a updateProposalWrappedStore, we retrieve the subjectStore
+		// because the contract call will be made on the client with the ID of the subjectStore
+		clientStore = upws.subjectStore
+	}
+
+	store, ok := clientStore.(storeprefix.Store)
```

Comment:
We are using [this Store](https://github.com/cosmos/cosmos-sdk/blob/store/v1.0.0-rc.0/store/prefix/store.go#L15-L21), this might help the reviewers